lib/engine-addon.js

```diff
@@ -358,14 +361,15 @@ module.exports = {
 
       // Replace `treeForAddon` so that we control how this engine gets built.
       // We may or may not want it to be combined like a default addon.
-      this.treeForAddon = function(engineSourceTree) {
+      this.treeForAddon = function() {
         if (this.lazyLoading === true) { return; }
 
         // NOT LAZY LOADING!
         // This is the scenario where we want to act like an addon.
-        var engineCSSTree = buildEngineCSSTree.call(this);
+        var engineCSSTree = buildEngineStyleTree.call(this);
```

Comment:
This is now returning the nested already-preprocessed child CSS trees via https://github.com/ember-cli/ember-cli/blob/16e4492c9ebf3348eb0f31df17215810674dbdf6/lib/models/addon.js#L781-L794

You split it to enable memoization. Can some of this end up documented in comments?